src/cc/lib/distributed/graph_engine.cc

```diff
@@ -110,13 +118,25 @@ grpc::Status GraphEngineServiceImpl::GetNodeFeatures(::grpc::ServerContext *cont
             continue;
         }
 
-        response->add_offsets(node_offset);
         response->mutable_feature_values()->resize(feature_offset + fv_size);
-        const auto index = internal_id->second;
+        auto index = internal_id->second;
+        size_t partition_count = m_counts[index];
         auto data = reinterpret_cast<uint8_t *>(response->mutable_feature_values()->data());
-        m_partitions[m_partitions_indices[index]].GetNodeFeature(m_internal_indices[index], features,
-                                                                 std::span(data + feature_offset, fv_size));
-        feature_offset += fv_size;
+        bool found = false;
+        for (size_t partition = 0; partition < partition_count && !found; ++partition, ++index)
+        {
+            found = m_partitions[m_partitions_indices[index]].GetNodeFeature(m_internal_indices[index], features,
+                                                                             std::span(data + feature_offset, fv_size));
```

Comment:
nit: take this variable `std::span(data + feature_offset, fv_size)` outside of the for loop.